core/src/main/scala/io/dantb/contentless/ContentModel.scala

```diff
@@ -63,12 +70,15 @@ object Field:
     a.fieldType === b.fieldType && a.defaultValue === b.defaultValue
   }
 
+// Include everything here: https://www.contentful.com/help/available-validations/
 sealed trait Validation extends Product with Serializable
 
 object Validation:
   given eq: Eq[Validation] = Eq.fromUniversalEquals
 
-  final case class ContainedIn(allowedValues: List[String]) extends Validation
+  final case class ContainedIn(allowedValues: NonEmptyList[String])        extends Validation
+  final case class ContainedInInt(allowedValues: NonEmptyList[Int])        extends Validation
+  final case class ContainedInDecimal(allowedValues: NonEmptyList[Double]) extends Validation
```

Comment:
This is what you posted the other day right? About erasure making you lose information unless you reify the type as a value, or do this.

Reply:
Yeah :( the problem is it's not a true GADT since the super type `Validation` doesn't expose that type parameter, so you'd end up having to do runtime inspection when pattern matching or something else more complex. I figured it's not exposed in the user API anyway so just do the verbose thing

Reply:
What would "reify type as a value" look like here?

Reply:
Something like

```scala
enum AcceptedTypes[T]:
  case int extends AcceptedTypes[Int]
  case string extends AcceptedTypes[String]
  ...
```

Then you could have something like

```scala
final case class ContainedIn[T](typ: AcceptedType[T], NonEmptyList[T])
```

Or alternatively you can fuse them both into

```scala
enum ContainedIn[T] extends Validation:
  case Int(allowedValues: NonEmptyList[Int]) extends ContainedIn[Int]
  ...
```

Which I suspect is better for pattern matching. But ultimately this is equivalent to what you were doing. Useful if you want to handle all types of `ContainedIn[T]` similarly though.